gateway/src/clickhouse_migration_manager/migrations/migration_0001.rs

```diff
@@ -0,0 +1,201 @@
+use std::time::Duration;
+
+use crate::clickhouse::ClickHouseConnectionInfo;
+use crate::clickhouse_migration_manager::migration_trait::Migration;
+use crate::error::Error;
+
+/// This migration is used to set up the ClickHouse database to efficiently validate
+/// the type of demonstrations.
+/// To do this, we need to be able to efficiently query what function was called for a
+/// particular inference ID.
+///
+/// As the original table was not set up to index on inference ID we instead need to create
+/// a materialized view that is indexed by inference ID and maps to the function_name that was used.
+/// Since we also would like to be able to get the original row from the inference ID we will keep the function_name,
+/// variant_name and episode_id in the materialized view so that we can use them to query the original table.
+
+pub struct Migration0001<'a> {
+    pub clickhouse: &'a ClickHouseConnectionInfo,
+    pub clean_start: bool,
+}
+
+impl<'a> Migration for Migration0001<'a> {
+    /// Check if you can connect to the database
+    async fn can_apply(&self) -> Result<(), Error> {
+        self.clickhouse
+            .health()
+            .await
+            .map_err(|e| Error::ClickHouseMigration {
+                id: "0001".to_string(),
+                message: e.to_string(),
+            })
+    }
```

Comment:
This should include verification for `0000` - e.g. checking if the two inference source tables exist.